spec/models/post_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:user) { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+
+  subject { user.posts.create(title: 'Coding in progress', text: 'One line at a time') }
+
+  describe 'validations' do
+    it 'should have a title' do
+      subject.title = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'should have text' do
+      subject.text = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'title should not be greater than 250' do
+      subject.title = 'Hello world' * 25
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments_counter should be an integer' do
+      subject.comments_counter = ''
+      expect(subject).to_not be_valid
+    end
+
+    it 'likes_counter should be an integer' do
+      subject.likes_counter = ''
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments_counter should be greater than or equal to 0' do
+      subject.comments_counter = -1
+      expect(subject).to_not be_valid
+    end
+
+    it 'likes_counter should be greater than or equal to 0' do
+      subject.likes_counter = -1
+      expect(subject).to_not be_valid
+    end
+  end
+
+  describe '#recent_comments' do
+    let(:commenter2) { User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.') }
+
+    it 'returns the most recent 5 comments' do
+      subject.comments.create(author: commenter2, text: 'Hi Tom!')
+      comment2 = subject.comments.create(author: user, text: 'Hi Lilly!')
+      comment3 = subject.comments.create(author: commenter2, text: 'Hope you all done with your project?')
+      comment4 = subject.comments.create(author: user, text: 'I asked after you from Ebun yesterday')
+      comment5 = subject.comments.create(author: commenter2, text: 'Oh, really, thanks.')
+      comment6 = subject.comments.create(author: user, text: 'I am almost done with the project')
+      recent_comments = subject.recent_comments
+
+      expect(recent_comments).to contain_exactly(comment6, comment5, comment4, comment3, comment2)
+      expect(recent_comments.length).to eq(5)
+    end
+  end
+
+  describe '#update_posts_counter' do
+    it 'updates the posts_counter on the associated author after save' do
+      user.update(posts_counter: 0)
+
+      expect do
+        user.posts.create(title: 'New Post', text: 'Content')
+        user.reload
+      end.to change { user.posts_counter }.by(1)
+    end
+  end
```

Comment:
- [ ] Good job trying to fix all the changes mentioned by the previous reviewer. However while testing your `model specs` I noticed that all your tests are failing to pass, and most of reasons I saw in the result is because of typo error. In many tests you wrote `posts_counter` whereas this column in your tables is named `post_counter`. I recommend you check everything and ensure that all your tests are passing before requesting a new review. Your specs to pass is the major requirements of this project.
![Screenshot from 2023-06-10 14-34-10](https://github.com/Tafloninno/Blog-App/assets/83042742/6f30fc50-9ce1-4d53-bf59-8b5b9a9f5192)

Reply:
Hello @tresorsawasawa 
<img width="1440" alt="Screenshot 2023-06-10 at 15 56 41" src="https://github.com/Tafloninno/Blog-App/assets/111200646/d85dd288-e440-4541-a546-889848773100">
This is  my last review and my tests are passing on my side